packages/pwa/app/utils/url.js

```diff
@@ -185,3 +196,67 @@ export const removeQueryParamsFromPath = (path, keys) => {
 
     return `${pathname}${paramStr && '?'}${paramStr}`
 }
+
+/**
+ * Rebuild the path with locale/site value to the path as url path or url query param
+ * based on url config
+ * @param {string} url - based url of the output url
+ * @param {object} configValues - object that contains values of url param config
+ * @return {string} - an output url
+ *
+ * @example
+ * //pwa-kit-config.json
+ * url {
+ *    locale: "query_param"
+ * }
+ * buildPathWithLocaleAndSiteParams('/women/dresses', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /women/dresses?locale=en-GB
+ *
+ *  @example
+ * //pwa-kit-config.json
+ * url {
+ *    locale: "path"
+ * }
+ * buildPathWithLocaleAndSiteParams('/women/dresses', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /en-GB/women/dresses
+ *
+ */
+export const buildPathWithUrlConfigParams = (url, configValues = {}) => {
+    const urlConfig = getUrlConfig()
+    if (!Object.values(configValues).length) return url
+    const {locale, siteAlias} = configValues
+    const queryParams = {}
+    const basePathSegments = []
+
+    const localeParamType = urlConfig['locale']
+    if (localeParamType === urlParamTypes.QUERY_PARAM) {
+        queryParams.locale = locale
+    }
+
+    if (localeParamType === urlParamTypes.PATH) {
+        basePathSegments.push(locale)
+    }
+
+    const siteAliasParamType = urlConfig['siteAlias']
+    if (siteAliasParamType === urlParamTypes.QUERY_PARAM) {
+        queryParams.locale = siteAlias
+    }
+
+    if (siteAliasParamType === urlParamTypes.PATH) {
+        basePathSegments.push(siteAlias)
+    }
```

Comment:
If you want to shorten up this code and make it a little more future proof, you could do something like: 

```
const options = ['locale', 'siteAlias']

options.forEach((option) => {
    const position = urlConfig['option'] || 'none'

    switch(position) {
        case urlParamTypes.QUERY_PARAM:
           queryParams[option]= configValues[option]
        case urlParamTypes.PATH:
           basePathSegments.push(configValues[option])
    }
})
```

This will do a few things, allow you to control the order the options are processed, expand if there are new options down the line.